File: sqry-lang-php/tests/fixtures/graph/users_controller.php
<?php

class UsersController {
    public function create() {
        $user = new User();
        $user->save();
        $this->sendWelcomeEmail();
    }

    public function sendWelcomeEmail() {
        Mailer::deliver();
    }

    public static function log($message) {
        Logger::info($message);
    }

    public static function audit($user) {
        self::log("audit: " . $user->id);
    }
}

File: sqry-lang-ruby/tests/fixtures/graph/users_controller.rb
class UsersController
  def create
    user = User.new
    user.save
    send_welcome_email
  end

  def send_welcome_email
    Mailer.deliver
  end

  def self.log(message)
    Logger.info(message)
  end

  def self.audit(user)
    log("audit: #{user.id}")
  end
end
